src/net.rs

```diff
@@ -348,6 +348,20 @@ pub fn tcp6() -> ProcResult<Vec<TcpNetEntry>> {
     read_tcp_table(BufReader::new(file))
 }
 
+/// Reads the specific tcp socket table
+pub fn tcp_with_pid(pid: i32) -> ProcResult<Vec<TcpNetEntry>> {
+    let file = FileWrapper::open(format!("/proc/{}/net/tcp", pid))?;
+
+    read_tcp_table(BufReader::new(file))
+}
```

Comment:
I'm still not sure how this new function is different from [Process::tcp()](https://docs.rs/procfs/0.14.1/procfs/process/struct.Process.html#method.tcp)